17 admin add user

andrewyang03 opened the pull request:
## Changes:
#11 17_admin-add-user

`CreateUser.tsx`
Created a create user account page that allows the user to create either a general account or admin account. 
- if not all fields filled out, an error pops up saying "Please fill out all required fields."
- on form submit, 
  - a popup confirmation displays 
  - calls addUser in firestoreFunctions.ts (from task 3) to add the user to the firestore database.

`CreateUser.module.css`

`popup.css` - styling for popup


![UserPage17](https://github.com/Hack4Impact-UMD/jefferson-area-board-for-aging/assets/108111995/7fa46a37-1cf0-4ec7-baf5-7bbb7470fa62)
_Note: Uses overflow so the create button is on the bottom_
![Popup17](https://github.com/Hack4Impact-UMD/jefferson-area-board-for-aging/assets/108111995/830ae5bf-8f90-434d-a1b1-206586ca671b)

andrewyang1015 committed:
17-admin_add_user
sample template for add user page

yumegan committed:
open CreateUser page, add check icon

yumegan committed:
remove priority & additional info

yumegan committed:
change button name to Create Profile

andrewyang1015 committed:
17-admin_add_user
allowed for popup to display User confirmation and Admin confirmation

andrewyang1015 committed:
Merge branch '17-admin_add_user' of https://github.com/Hack4Impact-UMD/jefferson-area-board-for-aging into 17-admin_add_user

andrewyang1015 committed:
17_admin_add_user changed header

yumegan committed:
fix css
button needs to be centered

yumegan committed:
Merge branch '17-admin_add_user' of https://github.com/Hack4Impact-UMD/jefferson-area-board-for-aging into 17-admin_add_user

andrewyang1015 committed:
17-admin_add_user
centered the submit button

andrewyang1015 committed:
17-admin_add_user
adjusted style changes to convention

yumegan committed:
remove urgency & additional info fields
from formData object

yumegan committed:
Merge branch '17-admin_add_user' of https://github.com/Hack4Impact-UMD/jefferson-area-board-for-aging into 17-admin_add_user

yumegan committed:
add user to database
with addUser function from firestoreFunctions.ts from task #3
create a userData object
add that to database

yumegan committed:
title corresponds to position

yieb commented on jaba/src/pages/CreateUserPage/CreateUser.tsx:
```diff
@@ -0,0 +1,204 @@
+import requestIconSuccess from '../../assets/requestIconSuccess.svg';
+import { addUser } from '../../backend/firestoreFunctions';
+
+const RequestAccountPage = () => {
```
Change function name to better reflect this page

yieb commented on jaba/src/pages/CreateUserPage/CreateUser.tsx:
```diff
@@ -0,0 +1,204 @@
+  }
+
+  return (
+    <div className={styles.requestPage}>
```
Update CSS style class names to better fit this page (rename so they are different from the class names used in the PR for Task 6)

yieb commented on jaba/src/pages/CreateUserPage/CreateUser.tsx:
```diff
@@ -0,0 +1,204 @@
+  );
+};
+
+export default RequestAccountPage;
```
Also change the export here

yieb commented on jaba/src/pages/CreateUserPage/CreateUser.module.css:
```diff
@@ -0,0 +1,250 @@
+body {
```
Update the styling a bit to match the edits I made to PR 6 before I merged it in: 
https://github.com/Hack4Impact-UMD/jefferson-area-board-for-aging/pull/6/commits/f341aace57fa95bb84d0301bd62fb1dd6860c8f9

(Reorder CSS classes in file so closely related CSS classes are closer together in the file, remove duplicate and dead CSS, update CSS to be a bit more responsive)

yumegan replied:
What do you mean by "update CSS to be a bit more responsive"?

yieb replied:
Missed this comment, sorry about that! It's been a while so I forget exactly what I was referring to anymore. I think I was referring to how  we want to make the frontend pages look better when the user zooms in or out, which would involve messing with padding and margin values and also trying to use units of measurement that are more responsive (so preferring using units like vw/vh and percentages instead of px values for sizes and padding/margin, etc).

yieb replied:
The updates do look better though!

yieb commented on jaba/src/pages/CreateUserPage/CreateUser.tsx:
```diff
@@ -0,0 +1,204 @@
+          <div className={styles.fields}>
+            <div className={styles.reqEntry} style={{ gridArea: 'fullname' }}>
+              <p className={styles.reqDescrip}>Full name</p>
+              <textarea
```
Change <textarea> tags to <input> and correct styling afterward

yumegan committed:
rename to CreateUser
instead of copied RequestAccountPage

yumegan committed:
change textarea to input, edit css field names
change field names to be more generic,
instead of being specific to req acc page.
change textarea to input to be single line instead of multi

yumegan committed:
fix css styling
keep body in bounds of white box
spacing / margin fixes

yumegan committed:
get rid of dead css
fix to match changes made to req acc page
not sure how to best place submit button

yumegan committed:
rearrange css to be more intuitive
group like elements together

yieb committed:
Fix bug w/button overlapping inputboxes on zoom in

yieb committed:
Remove debug CSS + rename CSS class for clarity

yieb committed:
Merge branch 'main' into 17-admin_add_user
